### ISSUE TITLE
Incorrect output fields names in docs

### ISSUE THREAD

**Issue by visheratin** [huggingface/transformers#7380]
## Environment info
     
- `transformers` version: 3.2.0
- Platform: Linux-5.4.0-7642-generic-x86_64-with-glibc2.29
- Python version: 3.8.2
- PyTorch version (GPU?): 1.6.0 (True)
- Tensorflow version (GPU?): not installed (NA)
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

### Who can help
@sgugger

## Information

The model I am using (Bert, XLNet ...): Bart

The problem arises when using the official example scripts.

```Python
from transformers import BartModel, BartTokenizer

tokenizer = BartTokenizer.from_pretrained('facebook/bart-base')
model = BartModel.from_pretrained('facebook/bart-base', return_dict=True, 
                                   output_hidden_states=True, output_attentions=True)

inputs = tokenizer("Hello, my dog is cute", return_tensors="pt")
outputs = model(**inputs)

print(outputs.hidden_states)
```

This script results in the error `'Seq2SeqModelOutput' object has no attribute 'hidden_states'`.

## To reproduce

1. Run the script above.

## Expected behavior

My expectation was to get a set of hidden states for the model. But in fact, the model returns two sets of hidden states - one for the decoder and another one for the encoder. It can be observed by looking at the keys of the `outputs`:

```Python
>>> print(outputs.keys())
odict_keys(['last_hidden_state', 'decoder_hidden_states', 'encoder_last_hidden_state', 'encoder_hidden_states'])
```

The same is valid for attentions if I specify `output_attentions=True`:

```Python
>>> print(outputs.keys())
odict_keys(['last_hidden_state', 'decoder_hidden_states', 'decoder_attentions', 'encoder_last_hidden_state', 'encoder_hidden_states', 'encoder_attentions'])
```

My conclusion is that the documentation gives an incorrect description of the output fields.

**Comment by visheratin** [comment]
Actually, the root of the problem might be related to the fact that the documentation states that the forward pass returns `` but in fact in returns `Seq2SeqModelOutput` ([source code](https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_bart.py#L947-L955)).

**Comment by sgugger** [comment]
Thanks for flagging! This should be fixed once the PR above is merged.

**Comment by LysandreJik** [comment]
Solved by #7390
